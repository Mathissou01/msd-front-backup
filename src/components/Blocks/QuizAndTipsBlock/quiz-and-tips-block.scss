@use "sass:list";
@use "sass:math";
@use "src/styles/01-settings/settings.colors" as colors;
@use "src/styles/01-settings/settings.blobs" as b;
@use "src/styles/02-tools/tools.blobs" as blobs;
@use "src/styles/02-tools/tools.include-media" as im;

.c-QuizAndTipsBlock {
  position: relative;
  display: flex;
  flex-direction: column;
  gap: 24px;
  align-items: flex-start;
  padding: 40px 16px 190px;
  overflow: hidden;
  background-color: colors.$white;
  border-bottom: 1px solid colors.$gray-200;

  @include im.media(">=mobile") {
    padding: 60px;
  }

  @include im.media(">=tablet") {
    gap: 60px;
    padding: 60px;
  }

  @include im.media(">=laptop") {
    padding: 80px 60px;
  }

  @include im.media(">=desktop") {
    padding: 80px 124px;
  }

  &__Tips {
    // Generate a random blob SVG with random rotation for any child element with "o-Blob" class
    $random-start: math.random(6);
    @for $i from 1 through (list.length(b.$blobs) * 2) {
      *:nth-child(#{(length(b.$blobs) * 2)}n + #{$i}) {
        // stylelint-disable-next-line
        .o-Blob::after {
          @include blobs.blob(
            (($random-start + $i) % list.length(b.$blobs)) + 1
          );

          transform: rotate(#{math.random(360deg)}deg);
        }
      }
    }

    display: flex;
    flex-direction: column;
    gap: 24px;
    align-items: center;
    width: 100%;

    @include im.media(">=tablet") {
      flex-direction: row;
      padding-right: 160px;
    }

    @include im.media(">=laptop") {
      padding-right: 102px;
    }
  }

  &__SvgContainer {
    position: absolute;
    right: 0;
    bottom: 0;
    display: flex;
    flex-direction: row;
    align-items: flex-end;
    justify-content: flex-end;
    width: 100%;
    height: 100%;
    pointer-events: none;

    svg {
      width: initial;
      height: 190px;

      @include im.media(">=mobile") {
        height: 254px;
        margin-right: -10px;
        margin-bottom: -10px;
      }

      #primary-color {
        fill: colors.$brand-primary;
      }

      #secondary-color {
        fill: colors.$brand-secondary;
      }
    }
  }

  &__Svg {
    &_mobile {
      display: initial;

      @include im.media(">=mobile") {
        display: none;
      }
    }

    &_desktop {
      display: none;

      @include im.media(">=mobile") {
        display: initial;
      }
    }
  }
}
