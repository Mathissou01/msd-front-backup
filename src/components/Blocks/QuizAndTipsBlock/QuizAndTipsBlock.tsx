import { QuizAndTipsBlockEntity } from "../../../graphql/codegen/generated-types";
import TipsMobile from "public/images/mobile_astuces-section_bottom-right-angle.svg";
import TipsDesktop from "public/images/desktop_astuces-section_bottom-right-angle.svg";
import CommonBlockHeading from "../../Common/CommonBlockHeading/CommonBlockHeading";
import TipCard from "./TipCard/TipCard";
import "./quiz-and-tips-block.scss";

interface IQuizAndTipsBlockProps {
  data: QuizAndTipsBlockEntity;
}

export default function QuizAndTipsBlock({ data }: IQuizAndTipsBlockProps) {
  /* Static Data */
  const labels = {
    knowMore: "En savoir plus",
  };

  /* Local Data */
  const titleContent = data.attributes?.titleContent ?? "";
  const quiz = data.attributes?.quiz?.data ?? null;
  const tips = data.attributes?.tips?.data ?? [];
  const hasQuiz = !!quiz;
  const hasTips = !!tips && tips.length > 0;

  return (
    <>
      {(hasQuiz || hasTips) && (
        <section className="c-QuizAndTipsBlock">
          <CommonBlockHeading titleContent={titleContent} />
          {/* // TODO: only display tips version for now, add quiz only and quiz+tips version later */}
          {/*{hasQuiz && <div />}*/}
          {hasTips && (
            <div className="c-QuizAndTipsBlock__Tips">
              {tips?.map((tip, index) => {
                if (tip.attributes?.shortDescription) {
                  return (
                    <TipCard
                      key={`tip_${tip.id}_${index}`}
                      tags={tip.attributes.tags?.data}
                      content={tip.attributes.shortDescription}
                      linkLabel={tip.attributes.link ?? labels.knowMore}
                      pictoUrl={
                        tip.attributes.image.data?.attributes?.url ?? null
                      }
                    />
                  );
                }
              })}
              <div className="c-QuizAndTipsBlock__SvgContainer">
                <TipsDesktop className="c-QuizAndTipsBlock__Svg_desktop" />
                <TipsMobile className="c-QuizAndTipsBlock__Svg_mobile" />
              </div>
            </div>
          )}
        </section>
      )}
    </>
  );
}
