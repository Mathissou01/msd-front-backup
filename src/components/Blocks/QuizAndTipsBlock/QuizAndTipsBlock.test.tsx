import { render, screen } from "@testing-library/react";
import {
  defaultMockData,
  threeTipsMockData,
} from "../../../../__mocks__/QuizAndTipsBlockMockData";
import QuizAndTipsBlock from "./QuizAndTipsBlock";

it("renders one tip", () => {
  const { container } = render(
    <QuizAndTipsBlock data={defaultMockData.data} />,
  );
  const tag = screen.getAllByText("Astuce");
  const firstTip = screen.getByText(
    "Découper vos peaux de bananes avant de les jeter dans votre compost.",
  );
  const secondTip = screen.queryByText(
    "Pas besoin de nettoyer vos pots de yaourts avant de les jeter",
  );
  expect(tag.length).toBe(1);
  expect(firstTip).toBeInTheDocument();
  expect(secondTip).not.toBeInTheDocument();
  expect(container).toMatchSnapshot();
});

it("renders three tips", () => {
  const { container } = render(
    <QuizAndTipsBlock data={threeTipsMockData.data} />,
  );
  const tag = screen.getAllByText("Astuce");
  const firstTip = screen.getByText(
    "Découper vos peaux de bananes avant de les jeter dans votre compost.",
  );
  const secondTip = screen.queryByText(
    "Pas besoin de nettoyer vos pots de yaourts avant de les jeter",
  );
  expect(tag.length).toBe(3);
  expect(firstTip).toBeInTheDocument();
  expect(secondTip).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
