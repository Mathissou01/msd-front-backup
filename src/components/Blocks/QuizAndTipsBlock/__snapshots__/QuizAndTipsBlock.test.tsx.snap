// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizAndTipsBlock renders one tip 1`] = `
<div>
  <section
    class="c-QuizAndTipsBlock"
  >
    <div
      class="c-CommonBlockHeading"
    >
      <h2>
        Astuces !
      </h2>
      <div
        class="c-CommonBlockHeading__Line"
      />
    </div>
    <div
      class="c-QuizAndTipsBlock__Tips"
    >
      <a
        class="c-TipCard"
        href="/"
      >
        <div
          class="c-TipCard__Tag"
        >
          Astuce
        </div>
        <p
          class="c-TipCard__Content"
        >
          Découper vos peaux de bananes avant de les jeter dans votre compost.
        </p>
        <span
          class="c-TipCard__Link"
        >
          En savoir plus
        </span>
        <div
          class="o-Blob c-TipCard__Svg"
        >
          <img
            alt=""
            data-nimg="1"
            decoding="async"
            height="72"
            loading="lazy"
            src="/images/pictos-temp/banana.svg"
            srcset="/images/pictos-temp/banana.svg 1x, /images/pictos-temp/banana.svg 2x"
            style="color: transparent;"
            width="72"
          />
        </div>
      </a>
      <div
        class="c-QuizAndTipsBlock__SvgContainer"
      >
        <div
          class="c-QuizAndTipsBlock__Svg_mobile"
        />
      </div>
    </div>
  </section>
</div>
`;

exports[`QuizAndTipsBlock renders three tips 1`] = `
<div>
  <section
    class="c-QuizAndTipsBlock"
  >
    <div
      class="c-CommonBlockHeading"
    >
      <h2>
        Astuces !
      </h2>
      <div
        class="c-CommonBlockHeading__Line"
      />
    </div>
    <div
      class="c-QuizAndTipsBlock__Tips"
    >
      <a
        class="c-TipCard"
        href="/"
      >
        <div
          class="c-TipCard__Tag"
        >
          Astuce
        </div>
        <p
          class="c-TipCard__Content"
        >
          Découper vos peaux de bananes avant de les jeter dans votre compost.
        </p>
        <span
          class="c-TipCard__Link"
        >
          En savoir plus
        </span>
        <div
          class="o-Blob c-TipCard__Svg"
        >
          <img
            alt=""
            data-nimg="1"
            decoding="async"
            height="72"
            loading="lazy"
            src="/images/pictos-temp/banana.svg"
            srcset="/images/pictos-temp/banana.svg 1x, /images/pictos-temp/banana.svg 2x"
            style="color: transparent;"
            width="72"
          />
        </div>
      </a>
      <a
        class="c-TipCard"
        href="/"
      >
        <div
          class="c-TipCard__Tag"
        >
          Astuce
        </div>
        <p
          class="c-TipCard__Content"
        >
          Pas besoin de nettoyer vos pots de yaourts avant de les jeter
        </p>
        <span
          class="c-TipCard__Link"
        >
          En savoir plus
        </span>
        <div
          class="o-Blob c-TipCard__Svg"
        >
          <img
            alt=""
            data-nimg="1"
            decoding="async"
            height="72"
            loading="lazy"
            src="/images/pictos-temp/sink.svg"
            srcset="/images/pictos-temp/sink.svg 1x, /images/pictos-temp/sink.svg 2x"
            style="color: transparent;"
            width="72"
          />
        </div>
      </a>
      <a
        class="c-TipCard"
        href="/"
      >
        <div
          class="c-TipCard__Tag"
        >
          Astuce
        </div>
        <p
          class="c-TipCard__Content"
        >
          Remplacez vos sacs en plastique par des sacs réutilisables, un caddie ou des paniers.
        </p>
        <span
          class="c-TipCard__Link"
        >
          En savoir plus
        </span>
        <div
          class="o-Blob c-TipCard__Svg"
        >
          <img
            alt=""
            data-nimg="1"
            decoding="async"
            height="72"
            loading="lazy"
            src="/images/pictos-temp/basket.svg"
            srcset="/images/pictos-temp/basket.svg 1x, /images/pictos-temp/basket.svg 2x"
            style="color: transparent;"
            width="72"
          />
        </div>
      </a>
      <div
        class="c-QuizAndTipsBlock__SvgContainer"
      >
        <div
          class="c-QuizAndTipsBlock__Svg_mobile"
        />
      </div>
    </div>
  </section>
</div>
`;
